src/cpu/core_dyn_x86/cache.h

```diff
@@ -156,7 +159,7 @@ class CodePageHandler : public PageHandler {
 			}
 			memset(invalidation_map,0,4096);
 		}
-		(*(Bit16u*)&invalidation_map[addr])+=0x101;
+		host_addw(&invalidation_map[addr], 0x0101);
```

Comment:
Need to use `add_to_unaligned_uint16` here to guarantee the same behavior as the original code on both big and little endian.

Sandbox here: https://onlinegdb.com/SkCUvfbA8
(flip the WORDS_BIGENDIAN on and off, and play with the `initial_val` and `add_val` values.)

Reply:
I'm glad you're bringing this up - please note this file is used only on little-endian (because it's core_dyn_x86); the equivalent line of code is in core_dynrec `CodePageHandlerDynRec::writew` function.

Overall CodePageHandler and CodePageHandlerDynRec are *the same* class; and their respective methods: `writeb`, `writew`, `writed`, `writeb_checked`, `writew_checked`, `writed_checked` are almost the same (next PR will remove the last difference they have).

So in this line in here (x86, little endian only) the code was: `(*(Bit16u*)&invalidation_map[addr])+=0x101;` while in dynrec (various archs, little and big endian) it was ifdef with separate code for little and big endian. I replaced both usages with `host_addw`, and I believe it to be correct (it's a step towards replacing 2 copies of `cache.h` with a single copy that works the same on both `core_dyn_x86` and `core_dynrec`.

Please compare these functions (between these two `cache.h` files) and double-check if my assumptions in here are correct.

Reply:
@dreamer , thanks for explaining the next goal to combining the cache files - it makes sense now so the function names overlap exactly (and are still correct in the x86 little-endian scenario for this commit, like you've mentioned). 

> Please compare these functions (between these two cache.h files) and double-check if my assumptions in here are correct.

I went through them side by side again, and it looks go.  I see now that this PR normalizes these two files, bringing them in-tandem with each in preparation for the big collapse into one!  This is so nice :-)